horovod/keras/callbacks.py

```diff
@@ -24,17 +38,11 @@ def __init__(self, root_rank, device=''):
             device: Device to be used for broadcasting. Uses GPU by default
                     if Horovod was build with HOROVOD_GPU_BROADCAST.
         """
-        super(BroadcastGlobalVariablesCallback, self).__init__()
-        self.root_rank = root_rank
-        self.device = device
-
-    def on_train_begin(self, logs=None):
-        with tf.device(self.device):
-            bcast_op = hvd.broadcast_global_variables(self.root_rank)
-            K.get_session().run(bcast_op)
+        _impl.BroadcastGlobalVariablesCallbackImpl.__init__(self, K, root_rank, device)
+        keras.callbacks.Callback.__init__(self)
```

Comment:
Should we initialize superclass first?  https://stackoverflow.com/a/15777488

Reply:
It's multiple inheritance, so both are superclasses.  But using this "old style" multiple inheritance was kinda ugly and error prone, so I switched it to "cooperative multiple inheritance".